libopendavinci/src/odcore/data/Container.cpp

```diff
@@ -253,6 +253,7 @@ namespace odcore {
                         // Check validity of the received bytes.
                         if (!( (0x0D == byte0) && (0xA4 == byte1) )) {
                             std::cerr << "[core::base::Container] Failed to decode OpenDaVINCI container header." << std::endl;
+                            return in;
```

Comment:
@fgiaimo This return ensure that in case of an incomplete OpenDaVINCI container, we won't try decoding content (would result in unknown behavior).